failed to build with gcc 11

gcc -v
```
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-pc-linux-gnu/11.1.0/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /build/gcc/src/gcc/configure --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=https://bugs.archlinux.org/ --enable-languages=c,c++,ada,fortran,go,lto,objc,obj-c++,d --with-isl --with-linker-hash-style=gnu --with-system-zlib --enable-__cxa_atexit --enable-cet=auto --enable-checking=release --enable-clocale=gnu --enable-default-pie --enable-default-ssp --enable-gnu-indirect-function --enable-gnu-unique-object --enable-install-libiberty --enable-linker-build-id --enable-lto --enable-multilib --enable-plugin --enable-shared --enable-threads=posix --disable-libssp --disable-libstdcxx-pch --disable-libunwind-exceptions --disable-werror gdc_include_dir=/usr/include/dlang/gdc
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 11.1.0 (GCC) 
```
error:
```
[  4%] Building C object CMakeFiles/soem.dir/soem/ethercatbase.c.o
[  9%] Building C object CMakeFiles/soem.dir/soem/ethercatcoe.c.o
[ 14%] Building C object CMakeFiles/soem.dir/soem/ethercatconfig.c.o
[ 19%] Building C object CMakeFiles/soem.dir/soem/ethercatdc.c.o
[ 23%] Building C object CMakeFiles/soem.dir/soem/ethercateoe.c.o
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c: In function ‘ecx_EOEsetIp’:
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:82:8: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
   82 |    EOEp->mbxheader.address = htoes(0x0000);
      |        ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:22: note: while referencing ‘MbxOut’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:83:8: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
   83 |    EOEp->mbxheader.priority = 0x00;
      |        ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:22: note: while referencing ‘MbxOut’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:90:8: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
   90 |    EOEp->mbxheader.mbxtype = ECT_MBXT_EOE + MBX_HDR_SET_CNT(cnt); /* EoE */
      |        ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:22: note: while referencing ‘MbxOut’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:92:8: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
   92 |    EOEp->frameinfo1 = htoes(EOE_HDR_FRAME_TYPE_SET(EOE_INIT_REQ) |
      |        ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:22: note: while referencing ‘MbxOut’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:95:8: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
   95 |    EOEp->frameinfo2 = 0;
      |        ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:22: note: while referencing ‘MbxOut’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:106:43: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  106 |       EOE_ip_uint32_to_byte(&ipparam->ip, &EOEp->data[data_offset]);
      |                                           ^~~~~~~~~~~~~~~~~~~~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:22: note: while referencing ‘MbxOut’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:30:15: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
   30 |    byte_ip[0] = eoe_ip4_addr4(ip); /* 4th octet */
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:22: note: while referencing ‘MbxOut’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:112:47: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  112 |       EOE_ip_uint32_to_byte(&ipparam->subnet, &EOEp->data[data_offset]);
      |                                               ^~~~~~~~~~~~~~~~~~~~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:22: note: while referencing ‘MbxOut’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:30:15: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
   30 |    byte_ip[0] = eoe_ip4_addr4(ip); /* 4th octet */
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:22: note: while referencing ‘MbxOut’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:118:56: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  118 |       EOE_ip_uint32_to_byte(&ipparam->default_gateway, &EOEp->data[data_offset]);
      |                                                        ^~~~~~~~~~~~~~~~~~~~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:22: note: while referencing ‘MbxOut’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:30:15: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
   30 |    byte_ip[0] = eoe_ip4_addr4(ip); /* 4th octet */
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:22: note: while referencing ‘MbxOut’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:124:47: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  124 |       EOE_ip_uint32_to_byte(&ipparam->dns_ip, &EOEp->data[data_offset]);
      |                                               ^~~~~~~~~~~~~~~~~~~~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:22: note: while referencing ‘MbxOut’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:30:15: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
   30 |    byte_ip[0] = eoe_ip4_addr4(ip); /* 4th octet */
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:22: note: while referencing ‘MbxOut’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:131:14: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  131 |       memcpy(&EOEp->data[data_offset], (void *)ipparam->dns_name, EOE_DNS_NAME_LENGTH);
      |              ^~~~~~~~~~~~~~~~~~~~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:22: note: while referencing ‘MbxOut’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:135:8: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  135 |    EOEp->mbxheader.length = htoes(EOE_PARAM_OFFSET + data_offset);
      |        ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:22: note: while referencing ‘MbxOut’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:136:8: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  136 |    EOEp->data[0] = flags;
      |        ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:22: note: while referencing ‘MbxOut’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:150:20: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  150 |          if ((aEOEp->mbxheader.mbxtype & 0x0f) == ECT_MBXT_EOE)
      |                    ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:15: note: while referencing ‘MbxIn’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |               ^~~~~
In file included from /home/share/makepkg/soem/src/SOEM/oshw/linux/oshw.h:18,
                 from /home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:18:
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:152:37: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  152 |             frameinfo1 = etohs(aEOEp->frameinfo1);
      |                                     ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercattype.h:540:21: note: in definition of macro ‘etohs’
  540 |   #define etohs(A) (A)
      |                     ^
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:15: note: while referencing ‘MbxIn’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |               ^~~~~
In file included from /home/share/makepkg/soem/src/SOEM/oshw/linux/oshw.h:18,
                 from /home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:18:
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:153:33: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  153 |             result = etohs(aEOEp->result);
      |                                 ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercattype.h:540:21: note: in definition of macro ‘etohs’
  540 |   #define etohs(A) (A)
      |                     ^
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:70:15: note: while referencing ‘MbxIn’
   70 |    ec_mbxbuft MbxIn, MbxOut;
      |               ^~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c: In function ‘ecx_EOEgetIp’:
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:193:8: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  193 |    EOEp->mbxheader.address = htoes(0x0000);
      |        ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:22: note: while referencing ‘MbxOut’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:194:8: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  194 |    EOEp->mbxheader.priority = 0x00;
      |        ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:22: note: while referencing ‘MbxOut’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:201:8: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  201 |    EOEp->mbxheader.mbxtype = ECT_MBXT_EOE + MBX_HDR_SET_CNT(cnt); /* EoE */
      |        ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:22: note: while referencing ‘MbxOut’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:203:8: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  203 |    EOEp->frameinfo1 = htoes(EOE_HDR_FRAME_TYPE_SET(EOE_GET_IP_PARAM_REQ) |
      |        ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:22: note: while referencing ‘MbxOut’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:206:8: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  206 |    EOEp->frameinfo2 = 0;
      |        ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:22: note: while referencing ‘MbxOut’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:208:8: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  208 |    EOEp->mbxheader.length = htoes(0x0004);
      |        ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:22: note: while referencing ‘MbxOut’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:209:8: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  209 |    EOEp->data[0] = flags;
      |        ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:22: note: while referencing ‘MbxOut’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |                      ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:222:20: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  222 |          if ((aEOEp->mbxheader.mbxtype & 0x0f) == ECT_MBXT_EOE)
      |                    ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:15: note: while referencing ‘MbxIn’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |               ^~~~~
In file included from /home/share/makepkg/soem/src/SOEM/oshw/linux/oshw.h:18,
                 from /home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:18:
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:224:37: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  224 |             frameinfo1 = etohs(aEOEp->frameinfo1);
      |                                     ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercattype.h:540:21: note: in definition of macro ‘etohs’
  540 |   #define etohs(A) (A)
      |                     ^
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:15: note: while referencing ‘MbxIn’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |               ^~~~~
In file included from /home/share/makepkg/soem/src/SOEM/oshw/linux/oshw.h:18,
                 from /home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:18:
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:225:38: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  225 |             eoedatasize = etohs(aEOEp->mbxheader.length) - 0x0004;
      |                                      ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercattype.h:540:21: note: in definition of macro ‘etohs’
  540 |   #define etohs(A) (A)
      |                     ^
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:15: note: while referencing ‘MbxIn’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |               ^~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:232:29: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  232 |                flags = aEOEp->data[0];
      |                             ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:15: note: while referencing ‘MbxIn’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |               ^~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:243:41: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  243 |                   EOE_ip_byte_to_uint32(&aEOEp->data[data_offset],
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:15: note: while referencing ‘MbxIn’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |               ^~~~~
In file included from /home/share/makepkg/soem/src/SOEM/soem/ethercat.h:22,
                 from /home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:19:
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:43:14: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
   43 |       byte_ip[0]); /* 4th octet */
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.h:40:26: note: in definition of macro ‘EOE_HTONL’
   40 | #define EOE_HTONL(x) ((((x) & 0x000000ffUL) << 24) | \
      |                          ^
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.h:60:31: note: in expansion of macro ‘EOE_MAKEU32’
   60 |    (ipaddr)->addr = EOE_HTONL(EOE_MAKEU32(a,b,c,d))
      |                               ^~~~~~~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:39:4: note: in expansion of macro ‘EOE_IP4_ADDR_TO_U32’
   39 |    EOE_IP4_ADDR_TO_U32(ip,
      |    ^~~~~~~~~~~~~~~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:15: note: while referencing ‘MbxIn’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |               ^~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:250:41: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  250 |                   EOE_ip_byte_to_uint32(&aEOEp->data[data_offset],
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:15: note: while referencing ‘MbxIn’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |               ^~~~~
In file included from /home/share/makepkg/soem/src/SOEM/soem/ethercat.h:22,
                 from /home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:19:
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:43:14: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
   43 |       byte_ip[0]); /* 4th octet */
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.h:40:26: note: in definition of macro ‘EOE_HTONL’
   40 | #define EOE_HTONL(x) ((((x) & 0x000000ffUL) << 24) | \
      |                          ^
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.h:60:31: note: in expansion of macro ‘EOE_MAKEU32’
   60 |    (ipaddr)->addr = EOE_HTONL(EOE_MAKEU32(a,b,c,d))
      |                               ^~~~~~~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:39:4: note: in expansion of macro ‘EOE_IP4_ADDR_TO_U32’
   39 |    EOE_IP4_ADDR_TO_U32(ip,
      |    ^~~~~~~~~~~~~~~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:15: note: while referencing ‘MbxIn’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |               ^~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:257:41: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  257 |                   EOE_ip_byte_to_uint32(&aEOEp->data[data_offset],
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:15: note: while referencing ‘MbxIn’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |               ^~~~~
In file included from /home/share/makepkg/soem/src/SOEM/soem/ethercat.h:22,
                 from /home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:19:
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:43:14: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
   43 |       byte_ip[0]); /* 4th octet */
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.h:40:26: note: in definition of macro ‘EOE_HTONL’
   40 | #define EOE_HTONL(x) ((((x) & 0x000000ffUL) << 24) | \
      |                          ^
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.h:60:31: note: in expansion of macro ‘EOE_MAKEU32’
   60 |    (ipaddr)->addr = EOE_HTONL(EOE_MAKEU32(a,b,c,d))
      |                               ^~~~~~~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:39:4: note: in expansion of macro ‘EOE_IP4_ADDR_TO_U32’
   39 |    EOE_IP4_ADDR_TO_U32(ip,
      |    ^~~~~~~~~~~~~~~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:15: note: while referencing ‘MbxIn’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |               ^~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:264:41: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  264 |                   EOE_ip_byte_to_uint32(&aEOEp->data[data_offset],
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:15: note: while referencing ‘MbxIn’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |               ^~~~~
In file included from /home/share/makepkg/soem/src/SOEM/soem/ethercat.h:22,
                 from /home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:19:
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:43:14: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
   43 |       byte_ip[0]); /* 4th octet */
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.h:40:26: note: in definition of macro ‘EOE_HTONL’
   40 | #define EOE_HTONL(x) ((((x) & 0x000000ffUL) << 24) | \
      |                          ^
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.h:60:31: note: in expansion of macro ‘EOE_MAKEU32’
   60 |    (ipaddr)->addr = EOE_HTONL(EOE_MAKEU32(a,b,c,d))
      |                               ^~~~~~~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:39:4: note: in expansion of macro ‘EOE_IP4_ADDR_TO_U32’
   39 |    EOE_IP4_ADDR_TO_U32(ip,
      |    ^~~~~~~~~~~~~~~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:15: note: while referencing ‘MbxIn’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |               ^~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:281:45: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  281 |                   memcpy(ipparam->dns_name, &aEOEp->data[data_offset], dns_len);
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:182:15: note: while referencing ‘MbxIn’
  182 |    ec_mbxbuft MbxIn, MbxOut;
      |               ^~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c: In function ‘ecx_EOEsend’:
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:329:8: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  329 |    EOEp->mbxheader.address = htoes(0x0000);
      |        ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:319:15: note: while referencing ‘MbxOut’
  319 |    ec_mbxbuft MbxOut;
      |               ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:330:8: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  330 |    EOEp->mbxheader.priority = 0x00;
      |        ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:319:15: note: while referencing ‘MbxOut’
  319 |    ec_mbxbuft MbxOut;
      |               ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:373:11: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  373 |       EOEp->mbxheader.length = htoes((uint16)(4 + txframesize)); /* no timestamp */
      |           ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:319:15: note: while referencing ‘MbxOut’
  319 |    ec_mbxbuft MbxOut;
      |               ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:374:11: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  374 |       EOEp->mbxheader.mbxtype = ECT_MBXT_EOE + MBX_HDR_SET_CNT(cnt); /* EoE */
      |           ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:319:15: note: while referencing ‘MbxOut’
  319 |    ec_mbxbuft MbxOut;
      |               ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:376:11: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  376 |       EOEp->frameinfo1 = htoes(frameinfo1);
      |           ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:319:15: note: while referencing ‘MbxOut’
  319 |    ec_mbxbuft MbxOut;
      |               ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:377:11: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  377 |       EOEp->frameinfo2 = htoes(frameinfo2);
      |           ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:319:15: note: while referencing ‘MbxOut’
  319 |    ec_mbxbuft MbxOut;
      |               ^~~~~~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c: In function ‘ecx_EOErecv’:
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:430:17: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  430 |       if ((aEOEp->mbxheader.mbxtype & 0x0f) == ECT_MBXT_EOE)
      |                 ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:411:15: note: while referencing ‘MbxIn’
  411 |    ec_mbxbuft MbxIn;
      |               ^~~~~
In file included from /home/share/makepkg/soem/src/SOEM/oshw/linux/oshw.h:18,
                 from /home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:18:
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:433:35: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  433 |          eoedatasize = etohs(aEOEp->mbxheader.length) - 0x00004;
      |                                   ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercattype.h:540:21: note: in definition of macro ‘etohs’
  540 |   #define etohs(A) (A)
      |                     ^
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:411:15: note: while referencing ‘MbxIn’
  411 |    ec_mbxbuft MbxIn;
      |               ^~~~~
In file included from /home/share/makepkg/soem/src/SOEM/oshw/linux/oshw.h:18,
                 from /home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:18:
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:434:34: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  434 |          frameinfo1 = etohs(aEOEp->frameinfo1);
      |                                  ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercattype.h:540:21: note: in definition of macro ‘etohs’
  540 |   #define etohs(A) (A)
      |                     ^
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:411:15: note: while referencing ‘MbxIn’
  411 |    ec_mbxbuft MbxIn;
      |               ^~~~~
In file included from /home/share/makepkg/soem/src/SOEM/oshw/linux/oshw.h:18,
                 from /home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:18:
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:435:34: error: array subscript ‘ec_EOEt[0]’ is partly outside array bounds of ‘ec_mbxbuft’ {aka ‘unsigned char[1487]’} [-Werror=array-bounds]
  435 |          frameinfo2 = etohs(aEOEp->frameinfo2);
      |                                  ^~
/home/share/makepkg/soem/src/SOEM/soem/ethercattype.h:540:21: note: in definition of macro ‘etohs’
  540 |   #define etohs(A) (A)
      |                     ^
/home/share/makepkg/soem/src/SOEM/soem/ethercateoe.c:411:15: note: while referencing ‘MbxIn’
  411 |    ec_mbxbuft MbxIn;
      |               ^~~~~
cc1: all warnings being treated as errors
make[2]: *** [CMakeFiles/soem.dir/build.make:132: CMakeFiles/soem.dir/soem/ethercateoe.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:154: CMakeFiles/soem.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

--- a/soem/ethercateoe.h
+++ b/soem/ethercateoe.h
@@ -16,10 +16,12 @@ extern "C"
 {
 #endif
 
-#include "ethercattype.h" 
+#include "ethercat.h"
 
-/* use maximum size for EOE mailbox data */
-#define EC_MAXEOEDATA EC_MAXMBX
+/* use maximum size for EOE mailbox data - mbxheader and 2x frameinfo */
+#define EC_MAXEOEDATA (EC_MAXMBX - (sizeof(ec_mbxheadert) +\
+                                    sizeof(uint16_t) +\
+                                    sizeof(uint16_t)))
 
 /** DNS length according to ETG 1000.6 */
 #define EOE_DNS_NAME_LENGTH  32

